pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java

```diff
@@ -1139,6 +1143,23 @@ public void writeBrokerDataOnZooKeeper(boolean force) {
         }
     }
 
+    /**
+     * sort bundles by load and select topK bundles for each broker.
+     * @return the number of bundles selected
+     */
+    private int selectTopKBundle() {
+        bundleArr.clear();
+
+        bundleArr.addAll(loadData.getBundleData().entrySet());
```

Comment:
Are these thread-safe?

Reply:
`org.apache.pulsar.broker.loadbalance.impl.ModularLoadManagerImpl#writeBundleDataOnZooKeeper` is executed in single thread.
https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java#L318-L319
But, the bundle-data may be updated in another thread, which may fail the sorting process. I will fix this problem.

Reply:
fixed.